homeassistant/components/amcrest/camera.py

```diff
@@ -160,6 +173,14 @@ def __init__(self, name, device, ffmpeg):
 
     # Entity property overrides
 
+    @property
+    def should_poll(self) -> bool:
+        """Return True if entity has to be polled for state.
+
+        False if entity pushes its state to HA.
+        """
+        return self._api.available and not self._update_succeeded
```

Comment:
Home Assistant does not support should_poll to be dynamically changed. It's either polled or it's not.

Reply:
Ok, changed the implementation.